std/concurrency.d

```diff
@@ -455,13 +479,27 @@ private template isSpawnable(F, T...)
  *  pointer indirection.  This is necessary for enforcing isolation among
  *  threads.
  */
+
 Tid spawn(F, T...)(F fn, T args)
 if (isSpawnable!(F, T))
 {
     static assert(!hasLocalAliasing!(T), "Aliases to mutable thread-local data not allowed.");
+    return _spawnnoref(false, fn, args);
+}
+
+Tid spawn(F, T...)(F fn, ref T args)
```

Comment:
Doesn't this mean that all arguments must be passed by ref? What if one wants to pass some by ref and some not?

Reply:
Yes they are all passed as ref, then when the function (fn) is called the arguments without ref will be passed as a value

Reply:
Why not `auto ref`?

Reply:
I had problems using auto ref but I don't remember exactly what, it's been too long
I think that today I reproach with auto ref to check again

Reply:
Ok the problem of using auto ref is that when calling spawn (F, T ...) (F fn, ref T args) with a rvalue this is passed as a value and is ok but then when spawn calls _spawn (F, T ...) (bool linked, F fn, auto ref T args) at this point the rvalue is passed as a reference

Here I gave an example of what happens: https://run.dlang.io/is/yMHf8r

Not well I know what could be done to avoid this, perhaps combine _spawn and spawn or manually generate templates to avoid passing as a reference in _spawn

Reply:
That's because after it's been bound to a name (a variable) it comes an lvalue. When dealing with `auto ref`, you need to use `std.functional.forward` to pass it along to another function.

Reply:
Ah nice to know, I didn't know the existence of forward, ok then I'll rewrite the PR
Thank you for teaching me something new